src/libs/metadata_view.c

```diff
@@ -745,7 +745,7 @@ static void _metadata_view_update_values(dt_lib_module_t *self)
 
       case md_exif_focus_distance:
         (void)g_strlcpy(text, NODATA_STRING, sizeof(text));
-        if(img->exif_focus_distance == UINT32_MAX)
+        if(img->exif_focus_distance == (float)UINT32_MAX)
```

Comment:
Are we 100% sure about the wanted behaviour here?

UINT32_MAX does not fit into a float...

Reply:
@kmilos The maximum number that can be represented as a float (which is at least 32 bits on modern systems) is 3.402823e+38.
UINT32_MAX is 4294967295U. Why do you think that UINT32_MAX does not fit into a float?

Reply:
Please try the UINT32_MAX -> float -> uint32_t roundtrip conversion yourself.

float has a limited precision (23 bits of mantissa and 8 bits for exponent), and cannot represent UINT32_MAX exactly. It does have a large range indeed, but the (relative) precision is limited: https://en.wikipedia.org/wiki/Single-precision_floating-point_format

Reply:
Hum... right! And so having `img->exif_focus_distance` as a float cannot work here. @victoryforce I'll revert the change to display infinity for now and we will devise a new approach. Maybe at the time the exif data is read, if we find `FFFFFFFF` set the `img->exif_focus_distance` to `FTL_MAX`? Kind of a hack but maybe the only possible approach for now. Thoughts?

Reply:
Yep, that's more sensible. Perhaps also check if 0/0 also means infinity in the exif spec.

Reply:
Sorry for the delay in replying, the recent heat hasn't been very conducive to productivity.

First of all, thank you all for your review and suggestions, it helps to find better and more reliable solutions.

> Perhaps also check if 0/0 also means infinity in the exif spec.

No, the Exif specification states that a zero in the numerator means an unknown distance.

> Maybe at the time the exif data is read, if we find FFFFFFFF set the img->exif_focus_distance to FTL_MAX? Kind of a hack but maybe the only possible approach for now. 

The big drawback of this proposal is that `it will only work for new imports`. Any images that have already been imported by users will continue to display an absurdly long focus distance instead of the correct "infinity". So I think it's too late to implement such a change now, although in the initial design I might have chosen it.

Now about why I disagree with the statement that "float cannot work", don't worry about the loss of precision and insist that my code was correct:

While float cannot represent UINT32_MAX exactly, this is not relevant. Yes, roundtrip conversion will not yield the same original value. But the only thing we care about is that the type conversion process performed when reading and storing the metadata is identical to the one that occurs on the right side of the comparison operator, `(float)UINT32_MAX`. If the two processes are identical, then they will give identical values ​​and a comparison is guaranteed to confirm their equality.

What happens in the process of reading metadata (in exif.cc)? We convert the value of this tag (SubjectDistance) with the toFloat function, which is defined in the exiv2 library. The conversion code boils down to:

  static_cast<float>(value_.at(n).first) / value_.at(n).second;

In our case, when the camera recorded a value for infinity, the numerator here is UINT32_MAX, converted to float, and the denominator will be 1. It is clear that dividing by 1 will not change anything, although to be completely precise, operations on type float include its conversion to double, but this process does not lose accuracy.

So we actually have the same conversion (`(float)UINT32_MAX`) and the comparison is reliable.

Reply:
Well, I didn't say it was not correct behaviour, I just asked are we sure because of this gotcha!

Perhaps >= some threshold is safer than exact match?

Reply:
> Perhaps >= some threshold is safer than exact match?

No, it's not safer.

Let's imagine that this field (img->exif_focus_distance) has an integer type. Will we need to use a fuzzy (+/- threshold) comparison instead of an exact one? Will it be necessary to check not the equality of values, but whether their difference does not exceed some threshold? Certainly not.

We recorded in img->exif_focus_distance the result of casting a certain integer value to a float type. Ok, it doesn't matter that this is a casting, generally speaking, the result of some transformation function that gives a non-random result.

We apply this function to UINT32_MAX and get a certain result. Exactly the same function is applied when reading an Exif tag. So, if UINT32_MAX was written in the SubjectDistance of the image, we will get ___the same result___ and their exact comparison will show the equality of values.

There is no need for a threshold here.

Reply:
It’s not as simple as that, because exif_focus_distance is stored in database, meaning it could have been calculated in totally different conditions (CPU architecture, toolchain, etc.). It’s probably though not a real problem, especially since this is just a cosmetic fix.

Reply:
That. And there is also a risk that the exiv2 library might have been built w/ a different toolchain than the dt binary. Can we be absolutely sure the UINT32_MAX rounding when casting to float will be identical between the two?

Reply:
@TurboGit So, I will create a new PR now with the proposed comparison. It would be wiser for me to discuss the safety of this particular comparison later rather than risk missing the release deadline. :)